packages/core/src/components/forms/numericInput.tsx

```diff
@@ -524,6 +524,23 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     private isFloatingPointNumericCharacter(char: string) {
         return NumericInput.FLOATING_POINT_NUMBER_CHARACTER_REGEX.test(char);
     }
+
+    private getStepMaxPrecision(props: HTMLInputProps & INumericInputProps) {
+        let stepMaxPrecision = Utils.countDecimalPlaces(props.stepSize);
+        if (props.minorStepSize != null) {
+            stepMaxPrecision = Math.max(stepMaxPrecision, Utils.countDecimalPlaces(props.minorStepSize));
+        }
+
+        return stepMaxPrecision;
```

Comment:
what if you did like:

```js
if (minorStepSize) { return countDecimalPlaces(minorStepSize); }
else { return countDecimalPlaces(stepSize); }
```

then we only count maximum once. and the `minorStepSize` should be the smallest one, right? so we can just go off that.